### DISCUSSION TITLE
QOL Improvement: Randomize shop item colors when placing a new shop.

### DISCUSSION THREAD

**Discussion by grahambilley** [OpenRCT2/OpenRCT2#15778] [Feature requests]
**OS:** [Windows 10]
**Version:** [0.2.4]
**Commit/Build:** [d645338]

This is an enhancement request. I'm putting this here per request of Aaron van Geffen.

Randomize the color of shop items (hats, balloons, shirts, umbrellas) when placing a new shop, instead of always defaulting to red (toggle on/off in settings).

Thank you,

**Comment by grahambilley** [comment]
Similar requests have been mentioned before. Referencing the other posts here. #1438/#2154

**Comment by Umdlye** [comment]
I deleted my earlier comment because I misread your suggestion, those two aren't the same as yours. They want shops to be able to sell a randomly colored item every sale. You just want the preset color to be different every time a shop is placed, right?

**Comment by grahambilley** [comment]
Thanks for clarifying. I suppose both suggestions solve the same problem. But I think having the default color be random is a better solution, since you still have the ability to color code areas of your park if you want to. If there was a "randomize item color" toggle where every sale was random, you would lose that ability.

**Comment by ocalhoun6** [comment]
Seems like this would be _perfect_ for the kind of a thing a plugin should do.

**Comment by pizza2004** [comment]
Yeah, I would like the random option to be built into the game, basically being a new color to select from the color menu, but this would work very well as a plugin that just changes the color randomly when you place the shop.

**Comment by grahambilley** [comment]
I just want to clarify one thing on my intent for this enhancement (so this is just my opinion of course).

What I'm trying to avoid is the extra time and clicks changing the color of an item when I place a new shop. If there were a "random" color square that I could click, I would still have to pick that square every time I place a new shop, so it doesn't solve my problem. At that point, I may as well just pick a color that I like.

**Comment by grahambilley** [comment]
That is awesome! Thank you!
